test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+        let(:result) { :ok }
+        let(:value) { "some-value" }
+
+        before(:each) do
+          allow(Yast::UI).to receive(:QueryWidget).
+            with(client.Id("0_0"), :Value).and_return(value)
+        end
+
+        context "and a path was specified" do
+          let(:ask) { base_ask.merge("path" => "users,0,gecos") }
+
+          it "values are saved into the Profile at the specified path" do
+            expect(Yast::Profile).to receive(:setElementByList).
+              with(["users", 0, "gecos"], value, profile).
+              and_call_original
+            client.askDialog
+          end
+        end
+
+        context "and a pathlist was specified" do
+          let(:ask) { base_ask.merge("pathlist" => ["users,1000,login", "groups,1000,name"]) }
+
+          it "saves value into Profile at the paths specified in the pathlist" do
+            expect(Yast::Profile).to receive(:setElementByList).
+              with(["users", 1000, "login"], value, profile).and_call_original
+            expect(Yast::Profile).to receive(:setElementByList).
+              with(["groups", 1000, "name"], value, profile).and_call_original
+            client.askDialog
+          end
+        end
+
+        context "and a file was specified" do
+          let(:file_path) { "/tmp/value" }
+          let(:ask) { base_ask.merge("file" => file_path) }
+
+          it "saves value in the file" do
+            expect(Yast::SCR).to receive(:Write).
+              with(Yast::Path.new(".target.string"), file_path, value).
+              and_return(true)
+            client.askDialog
+          end
+
+          context "and value is a boolean" do
+            let(:value) { true }
+            let(:ask) { base_ask.merge("type" => "boolean", "file" => file_path)}
+
+            it "save the value in the file" do
+              expect(Yast::SCR).to receive(:Write).
+                with(Yast::Path.new(".target.string"), file_path, "true").
+                and_return(true)
+              client.askDialog
+            end
+          end
+        end
+
+        context "and a script was specified" do
+          let(:script) { { "source" => "echo", "filename" => "test.sh" } }
+          let(:ask) { base_ask.merge("script" => script) }
+          let(:tmp_dir) { "/tmp" }
+          let(:log_dir) { "/var/log/YaST2" }
+
+          before(:each) do
+            allow(Yast::AutoinstConfig).to receive(:tmpDir).and_return(tmp_dir)
+            allow(Yast::AutoinstConfig).to receive(:logs_dir).
+              and_return(log_dir)
+            script_path = "#{tmp_dir}/#{script["filename"]}"
+            allow(Yast::SCR).to receive(:Write).
+              with(Yast::Path.new(".target.string"), script_path, script["source"]).
+              and_return(true)
+            allow(Yast::SCR).to receive(:Execute).
+              with(Yast::Path.new(".target.mkdir"), File.join(tmp_dir, "ask_scripts_log"))
+          end
+
+          context "when environment is not set" do
```

Comment:
It's just a property of the question (ask). I've done it more clear but just changing the description to `when 'environment' property is not set`.